src/lib/xmlparse.rs

```diff
@@ -72,11 +72,14 @@ use std::alloc::{self, Layout};
 use std::cell::{Cell, RefCell};
 use std::cmp;
 use std::collections::{HashMap, HashSet, TryReserveError};
-use std::convert::{TryFrom, TryInto};
+use std::convert::TryInto;
 use std::mem;
-use std::ops;
+use std::ops::{self, DerefMut};
 use std::ptr;
-use std::rc::Rc;
+use std::rc::{Rc, Weak};
+use std::slice;
+
+type Ptr<T> = Option<Rc<T>>;
```

Comment:
If I hadn't seen this alias, it wouldn't have been obvious what `Ptr<T>` is. `OptRc<T>` would be more obvious, but at that point it's not far off from `Option<Rc<T>>`. Is the full type that verbose that it warrants an alias?